src/deluge/gui/ui/sound_editor.cpp

```diff
@@ -916,8 +917,32 @@ ActionResult SoundEditor::potentialShortcutPadAction(int32_t x, int32_t y, bool
 			goto doSetup;
 		}
 
-		else {
+		// Name shortcut setup
+		else if (getCurrentClip()->type == ClipType::INSTRUMENT) {
+			if (x == 11 && y == 5) {
+
+				// Renames the clip for instrument SYNTH or KIT
+				Clip* clip = getCurrentClip();
+				Output* output = getCurrentOutput();
+
+				// Rename clip always for Synth & Midi Out, for KIT clipname can be altered when Effect entire is on to
+				// keep it consistent
+				if (output->type == OutputType::SYNTH || output->type == OutputType::MIDI_OUT
+				    || output->type == OutputType::KIT && getRootUI()->getAffectEntire()) {
+
+					if (clip) {
+						renameClipNameUI.clip = clip;
+						openUI(&renameClipNameUI);
+						return ActionResult::DEALT_WITH;
+					}
+				}
+			}
+			// if else then just keep it like the old way to rename the kit row item
+			goto doNextAfterName;
```

Comment:
this is in-extensible (nobody can add another condition to the if else chain without breaking the goto)

Could you put all the conditions into the if statement? Split this block out to a function that returns true if the editing took place and false otherwise so it continues down the chain. Then just return dealt with if the condition is met

Reply:
I also don't see why audio clips are excluded

Reply:
Agreed, was a bit of a dirty patch here with the label. Will warp it in a function.